src/lib/containers/MarkdownSynapse.js

```diff
@@ -212,242 +176,101 @@ class MarkdownSynapse extends React.Component {
-     *
-     * @param {*} elementList dictionary of 
-     * @memberof Markdown
-     */
-    matchElementToResource(elementList) {
-        let referenceCount = 0
-        let savedReferences = this.state.savedReferences || []
-
-        if (elementList.length > 0) {
-            elementList.forEach((elementBundle) => {
-                if (elementBundle.widgetType === "image") {
-                    // match corresponds to filehandle that this current element needs to be connected to
-                    let match = this.matchToHandle(this.compareById(elementBundle.id, "fileHandleId"), this.state.fileResults);
-                    this.handleImageWidget(match, elementBundle);
-                } else if (elementBundle.widgetType === "reference") {
-                    this.handleReferenceWidget(elementBundle, referenceCount++, savedReferences);
-                }
-            });
-            if (referenceCount > 0 && !this.state.hasProcessedReferences) {
-                this.setState({hasProcessedReferences: true, savedReferences: savedReferences, hasBookmarks: true})    
-            }
-        }
+    renderSynapseButton(widgetparamsMapped, element) {
+        let button = "<a href=\"" + widgetparamsMapped.url + "\"class=\"btn btn-lg btn-info\" role=\"button\" >" + widgetparamsMapped.text + "</a>";
+        element.outerHTML = button;
     }
 
-    handleImageWidget(match, elementBundle) {
-        if (elementBundle.element.parentElement) {
-            let renderedHTML = `<image class="img-fluid" src=${match[0].preSignedURL}></image>`;
-            elementBundle.element.outerHTML = renderedHTML;
-        }
-    }         
+    renderSynapsePlot(widgetparamsMapped, element) {
+        let plotWidget = <SynapsePlot token={this.props.token} ownerId={this.props.ownerId} wikiId={this.props.wikiId} widgetparamsMapped={widgetparamsMapped} />;
+        ReactDOM.render(plotWidget, element);
+    }
 
-    /**
-     * Handles ?{reference} synapse widgets
-     *
-     * @param {*} elementBundle
-     * @param {*} index
-     * @memberof MarkdownSynapse
-     */
-    handleReferenceWidget(elementBundle, index, savedReferences) {
-        let renderedHTML = ""
-        // due to re-renering, we save the result of this method, on initial calculate the html, otherwise we just 
-        // show it
-        if (!this.state.hasProcessedReferences) {
-            renderedHTML = `<span> <span><div class="ReferenceWidget"><a href="#" class="margin-left-5">[${elementBundle.widgetparamsMapped.footnoteId}]</a></div></span></span>`
-            savedReferences.push(renderedHTML)
-        } else {
-            renderedHTML = savedReferences[index]
+    renderSynapseImage(widgetparamsMapped, element) {
+        if (widgetparamsMapped.fileName) {
+            let img = <SynapseImage token={this.props.token} fileName={widgetparamsMapped.fileName} wikiId={this.props.wikiId} isAttachedToEntity={false} fileResults={this.state.fileHandles.list} />;
```

Comment:
Can remove isAttachedToEntity?  If synapse id is set, then use it.  And if wiki page id is set, look in file list for file name.